InsertData/testsrc/InsertDataTest.java

```diff
@@ -0,0 +1,190 @@
+import static com.mongodb.client.model.Filters.in;
+import static org.junit.Assert.*;
+
+import org.bson.Document;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import com.mongodb.MongoClient;
+import com.mongodb.MongoClientURI;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.result.DeleteResult;
+
+import org.json.JSONException;
+import org.json.simple.parser.ParseException;
+import org.junit.Test;
+
+
+public class InsertDataTest {
+
+	String CONNECTION = "mongodb+srv://m001-student:student123#@sandbox-trhqa.mongodb.net/test";
+	
+	 
+	//Check the connection
+	@Test
+	public void TestConnection() {		
+		MongoClientURI clientUri = new MongoClientURI(CONNECTION);
+		try(MongoClient client = new MongoClient(clientUri))
+		{
+			assertNotNull(client);
+		}    
+	}
+	
+	//Test for adding one data set
+	@Test
+	public void AddOneOrderWithDataTest() {
+		
+		String sku = "MDBTS110";
+		String item = "color pencil";
+		Double unit_price = 9.50;
+		int quantity = 32;
+		
+		InsertData insertData = new InsertData();		
+		List<Document> filteredDocumentsBefore = insertData.CountOrderSize(CONNECTION);
+		InsertData.AddOneOrderWithData(CONNECTION, sku, item, unit_price, quantity);
+		List<Document> filteredDocumentsAfter = insertData.CountOrderSize(CONNECTION);
+			
+		assertEquals(filteredDocumentsBefore.size()+1, filteredDocumentsAfter.size());	
+	}
+
+	
+	@Test(expected=IllegalArgumentException.class)
+	public void AddOneOrderWithDataTest_InvalidConnectionString_ThrowsException() 
+	{
+		InsertData insertData = new InsertData();	
+		List<Document> AddDocuments = InsertData.AddOneOrderWithData("", "MDBTS120", "color pencil", 9.50, 32 );
+	}
+	
+	@Test(expected=IllegalArgumentException.class)
+	public void AddOneOrderWithDataTest_emptyValue_ThrowsException() 
+	{
+		InsertData insertData = new InsertData();	
+		List<Document> AddDocuments = InsertData.AddOneOrderWithData(CONNECTION, "", "color pencil", 0.0, 32 );
+	}
+	
+	
+	@Test
+	public void AddOneOrderWithFileTest() throws JSONException, IOException, ParseException {
+		
+		InsertData insertData = new InsertData();		
+		List<Document> filteredDocumentsBefore = insertData.CountOrderSize(CONNECTION);
+		List<Document> AddDocuments = insertData.AddOneOrderWithFile(CONNECTION,"simpledata.json");
```

Comment:
Since you are reading the file from the common place where compiled code resides, you could mention that so that reader can do the same.

Reply:
I will mention this is the markdown